src/core/server/integration_tests/saved_objects/migrations/zdt_2/sor_higher_version_docs.test.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { pick, range } from 'lodash';
+import Path from 'path';
+import fs from 'fs/promises';
+import { type TestElasticsearchUtils } from '@kbn/core-test-helpers-kbn-server';
+import '../jest_matchers';
+import { ISavedObjectsRepository } from '@kbn/core-saved-objects-api-server';
+import { SavedObjectsModelVersionMap } from '@kbn/core-saved-objects-server';
+import { getKibanaMigratorTestKit, startElasticsearch } from '../kibana_migrator_test_kit';
+import { delay, createType } from '../test_utils';
+import { getBaseMigratorParams } from '../fixtures/zdt_base.fixtures';
+import { SavedObjectsBulkCreateObject } from '@kbn/core-saved-objects-api-server';
+
+export const logFilePath = Path.join(__dirname, 'sor_higher.test.log');
+
+describe('Higher version doc conversion', () => {
+  let esServer: TestElasticsearchUtils['es'];
+  let repositoryV1: ISavedObjectsRepository;
+  let repositoryV2: ISavedObjectsRepository;
+
+  const getTestType = ({ includeVersion2 }: { includeVersion2: boolean }) => {
+    const modelVersions: SavedObjectsModelVersionMap = {
+      1: {
+        changes: [],
+        schemas: {
+          forwardCompatibility: (attrs: any) => {
+            return pick(attrs, 'text', 'bool');
+          },
+        },
+      },
+    };
+
+    if (includeVersion2) {
+      modelVersions[2] = {
+        changes: [
+          {
+            type: 'data_backfill',
+            transform: (document) => {
+              document.attributes.newField = 'someValue';
+              return { document };
+            },
+          },
+        ],
+        schemas: {
+          forwardCompatibility: (attrs: any) => ({ ...attrs }),
```

Comment:
not necessary for the test, but for completeness sake should we use the following?
```
return pick(attrs, 'text', 'bool', 'newField');
```

Reply:
Yeah, I admit I just got lazy here